metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -327,6 +328,10 @@ public ControllerResult<BrokerRegistrationReply> registerBroker(
         }
         int brokerId = request.brokerId();
         BrokerRegistration existing = brokerRegistrations.get(brokerId);
+        if (version < 2 || existing == null || request.previousBrokerEpoch() != existing.epoch()) {
```

Comment:
This can be addressed later when we implement KIP-966. Since ELR is a new concept, it seems that we only want to update it if the BrokerRegistration request is on the latest version?

Reply:
Sounds good. If the version is < 2, we can skip handling it because ELR can't be enabled. We can adjust it later.